client/service/ledger_validation.go

```diff
@@ -121,22 +121,19 @@ func (s ClientService) ValidateLedger(args ...interface{}) (result model.LedgerV
 			if !s.clientConfig.IsAuditorEnabled {
 				r <- nil
 				e <- nil
-			}
+			} else {
+				auditor := rpc.NewAuditorClient(s.auditorConnection)
+				trailer := metadata.MD{}
 
-			auditor := rpc.NewAuditorClient(s.auditorConnection)
-			trailer := metadata.MD{}
+				response, err := auditor.ValidateLedger(context.Background(), request, grpc.Trailer(&trailer))
 
-			response, err := auditor.ValidateLedger(context.Background(), request, grpc.Trailer(&trailer))
+				if err != nil && trailer.Len() > 0 {
+					err = getClientErrorFromTrailer(trailer)
+				}
 
-			if err != nil && trailer.Len() > 0 {
-				err = getClientErrorFromTrailer(trailer)
+				r <- response
+				e <- err
```

Comment:
This block needs to be put into `if else` to split the case of auditor-enabled or auditor-disabled mode.